src/hocs/withContract.js

```diff
@@ -0,0 +1,41 @@
+import React, {Component} from "react";
+import PropTypes from "prop-types";
+
+const withContract = (contract, mapContractToProps) => WrappedComponent =>
+  class WithContract extends Component {
+    static contextTypes = {
+      contracts: PropTypes.object.isRequired,
+      cacheNewContract: PropTypes.func.isRequired
+    };
+
+    componentDidMount() {
+      // If the contract doesn't exist in the cache then call back to have it resolved and cached.
+      if (!this.context.contracts[contract.contractName]) {
+        this.context.cacheNewContract(contract);
+      }
+    }
+
+    render() {
+      let contractProps = {};
+      if (this.context.contracts[contract.contractName]) {
```

Comment:
You do this lookup several times in this method, so it probably makes sense to put it in a variable (or a getter function.)

Let's say you store it as `contract`, it looks like your `resolved` prop would just be `!!contract`, which also prevents you from needing to clone the contract object.